pyfar/dsp/dsp.py

```diff
@@ -116,9 +115,10 @@ def group_delay(signal, frequencies=None, method='fft'):
         raise ValueError(
             "Invalid method, needs to be either 'scipy' or 'fft'.")
 
-    # flatten in numpy fashion if a single channel is returned
-    if signal.cshape == (1, ):
-        group_delay = np.squeeze(group_delay)
+    # squeeze last dimension. If only one frequency is given, shape (1, ) is
+    # returned instead of (1, 1)
+    if frequencies is not None and frequencies.size == 1:
+        group_delay = np.squeeze(group_delay, axis=-1)
```

Comment:
```suggestion
```
not sure why we squeeze the last dimension.

Reply:
Probably not really nescessary, but some tests failed without it and i thought it would be nicer like that.

With the squeeze we get rid of the singular frequency dimension, similar to when we squeeze singular filter channels in the filter classes. Otherwise a single channel with a single frequency given would be returned as an array of shape (1, 1).

Reply:
I can remove it. 

I think some tests failed without it and i thought it would be similar to when we squeeze singular filter dimensions in the filter classes.

Reply:
We decided not to squeeze and adopt the tests.